components/filter/filter-dimension-set.js

```diff
@@ -10,6 +10,13 @@ class FilterDimensionSet extends LitElement {
 
 	static get properties() {
 		return {
+			/**
+			 * Whether to automatically handle clearing or send an event for this dimension
+			 * This should only be set to "manual" for use cases with specific perf enhacements in place for this
+			 * Note: If you choose "manual" for your `search-type`, your `clear-type` will also be "manual" because we cannot update the values removed by searching
+			 * @type {'automatic'|'manual'}
+			 */
+			clearType: { type: String, attribute: 'clear-type' },
```

Comment:
Just to confirm: there are known valid cases where they'd want a manual search-type but an automatic clear-type or vice versa? Just want to make sure we can't merge these into one.

Reply:
Yeah, the one I know of is Quick Eval - they are fine with the "automatic" searching, but would want the clear events to apply their HM "clear" action instead of individually unapplying each filter.  They could do that but then they're not taking advantage of that whole functionality of the api, and I worry about perf (which was a problem back in the day at least).

One thing I hadn't really considered, but is technically an option - we could have them do a check for whether they can use the clear action for that dimension.  They'd look at all the events they got back, and see if any would fully clear a dimension.  That still doesn't solve things for those using `manual` search though, and if we have to handle sending clear events anyways...

Reply:
So _another_ thing I was considering was adjusting the structure of the `changes` array in the change event to be grouped by their dimension, and I _could_ add something like `cleared: true|false`.  That way you don't sign up for clear events - if you use manual search you'll _have_ to look for them, and if you want to optimize you can check for that value.

I think in that case I'd need to make certain to send off any applied filter events first in their own event.  Or maybe something like this?
```
detail: {
  changes: [
    { dimension: role, allCleared: true, cleared: [ 'admin', 'instructor'] },
    { dimension: course, allCleared: false, cleared: [ 'biology'], applied: ['art] }
  ]
}
```

Reply:
That definitely sounds interesting! Mostly I was just poking at it since having the two types _seems_ complicated to wire up, but if they both end up being necessary that's cool.

Reply:
It's definitely more complicated than I wanted, I went back and forth a lot while developing trying to make it less so.  Figured I'd spend the afternoon poking at the new idea of just including this info in the events.  This PR includes both clear and clear all: https://github.com/BrightspaceUI/core/pull/1640

It's definitely less code, and there's no new attributes introduced.  The bulk of the "logic" shifts from being about the different possible combinations with search + clear + clear-all, and moves to making sure the change event is accurate and sent immediately after processing.